### DISCUSSION TITLE
Timezone and daylight saving.

### DISCUSSION THREAD

**Discussion by fishjoe** [micropython/micropython#9730] [General]
Good Evening Experts, 

It's 19:21mp here in NZ with daylights saving however I found on my pico w, it shows 8:13pm? It looks like its not considering daylight saving, can this be fixed?

Also I'm trying to build a automated update script. By hosting a web server from the folder on laptop, holding files, then automatically updating them to pico using "request" download after testing the mod time. however, the mod time has been turned into gmt. is there a way I can read it back to local time? preferably considering daylight saving. 

many thanks.

**Comment by davefes** [comment]
Have a read through this: 
https://forum.micropython.org/viewtopic.php?f=16&t=3675
page 2 and page 3 have several suggestions.

I have to do a manual entry once a year in my implementation, which I will post if you can't get something working.

davef

**Reply by fishjoe** [reply]
This works thanks. Us maths calculations

**Comment by ThinkTransit** [comment]
Have a look at the datetime package in micropython-lib, it can handle DST.

[](https://github.com/micropython/micropython-lib/blob/master/python-stdlib/datetime)

There are examples in the test_datetime.py file on how to create a timezone with DST.

If you are using the latest version of micropython you can install with mip

```python
import mip
mip.install('datetime')
```
